docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1015 @@
+
+<p align="center">
+  <img src="UML/AddChapterCommandClass.png" width="800" alt="Class Diagram of add chapter command"/>
+  <br/>Figure <>. Class Diagram of add chapter command  
+</p>
+
+Given below is an example usage scenario at Module level and how the add chapter feature behaves at each step:
+
+Step 1: The user is currently in module level.
+
+Step 2: The user want to add a new chapter chapter1 and enters `add chapter1` command.
+
+Step 3: The `Kaji` object passes the user's command to the `Parser` object. Then `Parser` creates a new `AddChapterCommand` object and return the `AddChapterCommand` object to `Kaji`.
+
+Step 4: The `Kaji` object uses the public `AddChapterCommand#excute()` to add new chapters into Kaji application.
+
+Step 5: The `AddChapterCommand` object creates a new `Chapter` object with the `Chapter#chapterName` of chapter1. 
+
+Step 6: The new `chapter1 : Chapter` object will be passed into the `AddChapterCommand#addChapter()` with `Access` and `Storage`.
+
+Step 7: The `AddChapterCommand#addChapter()` gets the current `Module` object via `Access#getModule()` method, then it adds the new `chapter1: Chapter` object into the `Module#ChapterList`. 
+
+Step 8: The new `Module` object with the `chapter1: Chapter` object is updated to `Access` via the method `Access#setModule(Module)`.
+
+Step 9: The new `Chapter` is updated in the folder by executing the `Storage#createChapter()`. A 'chapter1' folder is being created under the 'data/module' folder.
+
+Step 10: The `AddChapterCommand#addChapter()` method returns a result message to the `AddChapterCommand#execute()` method which uses `Ui#showToUser()` to print the success message.
+
+The following diagram shows how the add chapter command feature works:
+
+<p align="center">
+  <img src="UML/AddChapterCommand.png" width="800" alt="Sequence Diagram of add chapter command"/>
+  <br/>Figure <>. Sequence Diagram of add chapter command  
+</p>
+
+#### 4.2.2. List Chapters Feature
+(Zeyu)
+
+The list chapters feature allows the user to list all chapters in module level 
+
+The list chapters mechanism is facilitated by `ListChaptersCommand`. It extends from the abstract class `ListCommand`. 
+
+In addition, it implements the following operations:
+* `ListChaptersCommand#execute()` — oversees entire execution for listing chapters.
+* `ListChaptersCommand#listChapters()` — gets and lists all chapters in module level.
+
+For instance, the user wants to list all chapters available in `CS2113T` (module name), a detailed description of what happens is shown below:
+
+* Step 1: The user is currently in `CS2113T` level. 
+
+* Step 2: The user enters `list` command to list all chapters in `CS2113T` level. 
+
+The following sequence diagram shows how the list chapters feature works:
+![Sequence Diagram of List Chapters](UML/listchap_seq_diagram.png)
```

Comment:
![image](https://user-images.githubusercontent.com/56807788/97667724-79d2d980-1abb-11eb-8ac9-05df208e7da8.png)
Should this diagram include activation bars?